aten/src/ATen/native/Blas.cpp

```diff
@@ -133,8 +133,19 @@ inline void dot_check(const Tensor& self, const Tensor& other) {
 }
 
 Tensor dot(const Tensor &self, const Tensor &other){
-  at::NoNamesGuard guard;
+  if (self.is_complex()) {
+    if (self.is_conj()) {
+      if (other.is_conj()) {
+        return (at::dot(self.conj(), other.conj())).conj();
+       } else {
+         return at::vdot(self.conj(), other);
+       }
+    } else if (other.is_conj()) {
+      return at::vdot(other.conj(), self);
```

Comment:
In all the cases, I believe it's better to call `at::native::vdot`, as it avoid a redispatch. Same for `at::native::dot`.

Reply:
yeah good catch!